abstract/group.go

```diff
@@ -51,6 +51,9 @@ type Secret interface {
 
 	// Set to a fresh random or pseudo-random secret
 	Pick(rand cipher.Stream) Secret
+
+	// SetBytes will take bytes and create a secret out of it
+	SetBytes([]byte) Secret
```

Comment:
Well, for the `(Un)Marshal` case, the snippet I posted above seems very normal and required to me, otherwise we get different representations of a secret ? 
And I've made the changes for the `Bytes()` now.